### PR TITLE
fix(organization): scope setActiveTeam to active organization

### LINKED ISSUE #9237: setActiveTeam doesn't validate if the team belongs to the organization thats active in the session
### Is this suited for github?

- [x] Yes, this is suited for github

### Reproduction

Steps to reproduce
1. Have a session with active organization.
2. Call setActive team with teamId from different organization
3. Results in a session with activeOrgId and with a teamId which doesn't belong to this org.

### Current vs. Expected behavior

setActiveTeam should be callable(or validated that) only with the teamId for the activeOrganization can be set or pass both organizationId and teamId in the request, either of one should be implemented.

### What version of Better Auth are you using?

1.6.2

### System info

```bash
{
  "system": {
    "platform": "darwin",
    "arch": "arm64",
    "version": "Darwin Kernel Version 25.4.0: Thu Mar 19 19:33:50 PDT 2026; root:xnu-12377.101.15~1/RELEASE_ARM64_T6050",
    "release": "25.4.0",
    "cpuCount": 18,
    "cpuModel": "Apple M5 Pro",
    "totalMemory": "48.00 GB",
    "freeMemory": "1.28 GB"
  },
  "node": {
    "version": "v24.14.1",
    "env": "development"
  },
  "packageManager": {
    "name": "npm",
    "version": "11.11.0"
  },
  "frameworks": [
    {
      "name": "react",
      "version": "19.2.5"
    },
    {
      "name": "hono",
      "version": "^4.12.12"
    }
  ],
  "databases": [
    {
      "name": "@prisma/client",
      "version": "^7.7.0"
    }
  ],
  "betterAuth": {
    "version": "^1.6.2",
    "config": {
      "secret": "[REDACTED]",
      "baseURL": "http://localhost:5150",
      "trustedOrigins": [
        "http://localhost:5504"
      ],
      "session": {
        "expiresIn": 172800,
        "updateAge": 86400,
        "cookieCache": {
          "enabled": true,
          "maxAge": 300
        }
      },
      "advanced": {
        "trustedProxyHeaders": true,
        "defaultCookieAttributes": {
          "sameSite": "Strict"
        },
        "cookiePrefix": "ops",
        "cookies": {
          "session_token": {
            "name": "auth",
            "attributes": {}
          }
        }
      },
      "account": {
        "accountLinking": {
          "allowDifferentEmails": false,
          "allowUnlinkingAll": true
        }
      },
      "emailVerification": {
        "autoSignInAfterVerification": true
      },
      "emailAndPassword": {
        "enabled": false
      },
      "socialProviders": {
        "github": {
          "clientId": "[REDACTED]",
          "clientSecret": "[REDACTED]"
        },
        "google": {
          "clientId": "[REDACTED]",
          "clientSecret": "[REDACTED]"
        }
      },

      "plugins": [
        {
          "name": "magic-link",
          "config": {
            "id": "magic-link",
            "version": "1.6.2",
            "endpoints": {},
            "rateLimit": [
              {
                "window": 60,
                "max": 5
              }
            ],
            "options": {}
          }
        },
        {
          "name": "organization",
          "config": {
            "id": "organization",
            "version": "1.6.2",
            "endpoints": {},
            "schema": {
              "organization": {
                "fields": {
                  "name": {
                    "type": "string",
                    "required": true,
                    "sortable": true
                  },
                  "slug": {
                    "type": "string",
                    "required": true,
                    "unique": true,
                    "sortable": true,
                    "index": true
                  },
                  "logo": {
                    "type": "string",
                    "required": false
                  },
                  "createdAt": {
                    "type": "date",
                    "required": true
                  },
                  "metadata": {
                    "type": "string",
                    "required": false
                  }
                }
              },
              "team": {
                "fields": {
                  "name": {
                    "type": "string",
                    "required": true
                  },
                  "organizationId": {
                    "type": "string",
                    "required": true,
                    "references": {
                      "model": "organization",
                      "field": "id"
                    },
                    "index": true
                  },
                  "createdAt": {
                    "type": "date",
                    "required": true
                  },
                  "updatedAt": {
                    "type": "date",
                    "required": false
                  }
                }
              },
              "teamMember": {
                "fields": {
                  "teamId": {
                    "type": "string",
                    "required": true,
                    "references": {
                      "model": "team",
                      "field": "id"
                    },
                    "index": true
                  },
                  "userId": {
                    "type": "string",
                    "required": true,
                    "references": {
                      "model": "user",
                      "field": "id"
                    },
                    "index": true
                  },
                  "createdAt": {
                    "type": "date",
                    "required": false
                  }
                }
              },
              "member": {
                "fields": {
                  "organizationId": {
                    "type": "string",
                    "required": true,
                    "references": {
                      "model": "organization",
                      "field": "id"
                    },
                    "index": true
                  },
                  "userId": {
                    "type": "string",
                    "required": true,
                    "references": {
                      "model": "user",
                      "field": "id"
                    },
                    "index": true
                  },
                  "role": {
                    "type": "string",
                    "required": true,
                    "sortable": true,
                    "defaultValue": "member"
                  },
                  "createdAt": {
                    "type": "date",
                    "required": true
                  }
                }
              },
              "invitation": {
                "fields": {
                  "organizationId": {
                    "type": "string",
                    "required": true,
                    "references": {
                      "model": "organization",
                      "field": "id"
                    },
                    "index": true
                  },
                  "email": {
                    "type": "string",
                    "required": true,
                    "sortable": true,
                    "index": true
                  },
                  "role": {
                    "type": "string",
                    "required": false,
                    "sortable": true
                  },
                  "teamId": {
                    "type": "string",
                    "required": false,
                    "sortable": true
                  },
                  "status": {
                    "type": "string",
                    "required": true,
                    "sortable": true,
                    "defaultValue": "pending"
                  },
                  "expiresAt": {
                    "type": "date",
                    "required": true
                  },
                  "createdAt": {
                    "type": "date",
                    "required": true
                  },
                  "inviterId": {
                    "type": "string",
                    "references": {
                      "model": "user",
                      "field": "id"
                    },
                    "required": true
                  }
                }
              },
              "session": {
                "fields": {
                  "activeOrganizationId": {
                    "type": "string",
                    "required": false
                  },
                  "activeTeamId": {
                    "type": "string",
                    "required": false
                  }
                }
              }
            },
            "$Infer": {
              "Organization": {},
              "Invitation": {},
              "Member": {},
              "Team": {},
              "TeamMember": {},
              "ActiveOrganization": {}
            },
            "$ERROR_CODES": {
              "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_NEW_ORGANIZATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_NEW_ORGANIZATION",
                "message": "You are not allowed to create a new organization"
              },
              "YOU_HAVE_REACHED_THE_MAXIMUM_NUMBER_OF_ORGANIZATIONS": {
                "code": "YOU_HAVE_REACHED_THE_MAXIMUM_NUMBER_OF_ORGANIZATIONS",
                "message": "You have reached the maximum number of organizations"
              },
              "ORGANIZATION_ALREADY_EXISTS": {
                "code": "ORGANIZATION_ALREADY_EXISTS",
                "message": "Organization already exists"
              },
              "ORGANIZATION_SLUG_ALREADY_TAKEN": {
                "code": "ORGANIZATION_SLUG_ALREADY_TAKEN",
                "message": "Organization slug already taken"
              },
              "ORGANIZATION_NOT_FOUND": {
                "code": "ORGANIZATION_NOT_FOUND",
                "message": "Organization not found"
              },
              "USER_IS_NOT_A_MEMBER_OF_THE_ORGANIZATION": {
                "code": "USER_IS_NOT_A_MEMBER_OF_THE_ORGANIZATION",
                "message": "User is not a member of the organization"
              },
              "YOU_ARE_NOT_ALLOWED_TO_UPDATE_THIS_ORGANIZATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_UPDATE_THIS_ORGANIZATION",
                "message": "You are not allowed to update this organization"
              },
              "YOU_ARE_NOT_ALLOWED_TO_DELETE_THIS_ORGANIZATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_DELETE_THIS_ORGANIZATION",
                "message": "You are not allowed to delete this organization"
              },
              "NO_ACTIVE_ORGANIZATION": {
                "code": "NO_ACTIVE_ORGANIZATION",
                "message": "No active organization"
              },
              "USER_IS_ALREADY_A_MEMBER_OF_THIS_ORGANIZATION": {
                "code": "USER_IS_ALREADY_A_MEMBER_OF_THIS_ORGANIZATION",
                "message": "User is already a member of this organization"
              },
              "MEMBER_NOT_FOUND": {
                "code": "MEMBER_NOT_FOUND",
                "message": "Member not found"
              },
              "ROLE_NOT_FOUND": {
                "code": "ROLE_NOT_FOUND",
                "message": "Role not found"
              },
              "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_NEW_TEAM": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_NEW_TEAM",
                "message": "You are not allowed to create a new team"
              },
              "TEAM_ALREADY_EXISTS": {
                "code": "TEAM_ALREADY_EXISTS",
                "message": "Team already exists"
              },
              "TEAM_NOT_FOUND": {
                "code": "TEAM_NOT_FOUND",
                "message": "Team not found"
              },
              "YOU_CANNOT_LEAVE_THE_ORGANIZATION_AS_THE_ONLY_OWNER": {
                "code": "YOU_CANNOT_LEAVE_THE_ORGANIZATION_AS_THE_ONLY_OWNER",
                "message": "You cannot leave the organization as the only owner"
              },
              "YOU_CANNOT_LEAVE_THE_ORGANIZATION_WITHOUT_AN_OWNER": {
                "code": "YOU_CANNOT_LEAVE_THE_ORGANIZATION_WITHOUT_AN_OWNER",
                "message": "You cannot leave the organization without an owner"
              },
              "YOU_ARE_NOT_ALLOWED_TO_DELETE_THIS_MEMBER": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_DELETE_THIS_MEMBER",
                "message": "You are not allowed to delete this member"
              },
              "YOU_ARE_NOT_ALLOWED_TO_INVITE_USERS_TO_THIS_ORGANIZATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_INVITE_USERS_TO_THIS_ORGANIZATION",
                "message": "You are not allowed to invite users to this organization"
              },
              "USER_IS_ALREADY_INVITED_TO_THIS_ORGANIZATION": {
                "code": "USER_IS_ALREADY_INVITED_TO_THIS_ORGANIZATION",
                "message": "User is already invited to this organization"
              },
              "INVITATION_NOT_FOUND": {
                "code": "INVITATION_NOT_FOUND",
                "message": "Invitation not found"
              },
              "YOU_ARE_NOT_THE_RECIPIENT_OF_THE_INVITATION": {
                "code": "YOU_ARE_NOT_THE_RECIPIENT_OF_THE_INVITATION",
                "message": "You are not the recipient of the invitation"
              },
              "EMAIL_VERIFICATION_REQUIRED_BEFORE_ACCEPTING_OR_REJECTING_INVITATION": {
                "code": "EMAIL_VERIFICATION_REQUIRED_BEFORE_ACCEPTING_OR_REJECTING_INVITATION",
                "message": "Email verification required before accepting or rejecting invitation"
              },
              "YOU_ARE_NOT_ALLOWED_TO_CANCEL_THIS_INVITATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_CANCEL_THIS_INVITATION",
                "message": "You are not allowed to cancel this invitation"
              },
              "INVITER_IS_NO_LONGER_A_MEMBER_OF_THE_ORGANIZATION": {
                "code": "INVITER_IS_NO_LONGER_A_MEMBER_OF_THE_ORGANIZATION",
                "message": "Inviter is no longer a member of the organization"
              },
              "YOU_ARE_NOT_ALLOWED_TO_INVITE_USER_WITH_THIS_ROLE": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_INVITE_USER_WITH_THIS_ROLE",
                "message": "You are not allowed to invite a user with this role"
              },
              "FAILED_TO_RETRIEVE_INVITATION": {
                "code": "FAILED_TO_RETRIEVE_INVITATION",
                "message": "Failed to retrieve invitation"
              },
              "YOU_HAVE_REACHED_THE_MAXIMUM_NUMBER_OF_TEAMS": {
                "code": "YOU_HAVE_REACHED_THE_MAXIMUM_NUMBER_OF_TEAMS",
                "message": "You have reached the maximum number of teams"
              },
              "UNABLE_TO_REMOVE_LAST_TEAM": {
                "code": "UNABLE_TO_REMOVE_LAST_TEAM",
                "message": "Unable to remove last team"
              },
              "YOU_ARE_NOT_ALLOWED_TO_UPDATE_THIS_MEMBER": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_UPDATE_THIS_MEMBER",
                "message": "You are not allowed to update this member"
              },
              "ORGANIZATION_MEMBERSHIP_LIMIT_REACHED": {
                "code": "ORGANIZATION_MEMBERSHIP_LIMIT_REACHED",
                "message": "Organization membership limit reached"
              },
              "YOU_ARE_NOT_ALLOWED_TO_CREATE_TEAMS_IN_THIS_ORGANIZATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_CREATE_TEAMS_IN_THIS_ORGANIZATION",
                "message": "You are not allowed to create teams in this organization"
              },
              "YOU_ARE_NOT_ALLOWED_TO_DELETE_TEAMS_IN_THIS_ORGANIZATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_DELETE_TEAMS_IN_THIS_ORGANIZATION",
                "message": "You are not allowed to delete teams in this organization"
              },
              "YOU_ARE_NOT_ALLOWED_TO_UPDATE_THIS_TEAM": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_UPDATE_THIS_TEAM",
                "message": "You are not allowed to update this team"
              },
              "YOU_ARE_NOT_ALLOWED_TO_DELETE_THIS_TEAM": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_DELETE_THIS_TEAM",
                "message": "You are not allowed to delete this team"
              },
              "INVITATION_LIMIT_REACHED": {
                "code": "INVITATION_LIMIT_REACHED",
                "message": "Invitation limit reached"
              },
              "TEAM_MEMBER_LIMIT_REACHED": {
                "code": "TEAM_MEMBER_LIMIT_REACHED",
                "message": "Team member limit reached"
              },
              "USER_IS_NOT_A_MEMBER_OF_THE_TEAM": {
                "code": "USER_IS_NOT_A_MEMBER_OF_THE_TEAM",
                "message": "User is not a member of the team"
              },
              "YOU_CAN_NOT_ACCESS_THE_MEMBERS_OF_THIS_TEAM": {
                "code": "YOU_CAN_NOT_ACCESS_THE_MEMBERS_OF_THIS_TEAM",
                "message": "You are not allowed to list the members of this team"
              },
              "YOU_DO_NOT_HAVE_AN_ACTIVE_TEAM": {
                "code": "YOU_DO_NOT_HAVE_AN_ACTIVE_TEAM",
                "message": "You do not have an active team"
              },
              "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_NEW_TEAM_MEMBER": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_NEW_TEAM_MEMBER",
                "message": "You are not allowed to create a new member"
              },
              "YOU_ARE_NOT_ALLOWED_TO_REMOVE_A_TEAM_MEMBER": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_REMOVE_A_TEAM_MEMBER",
                "message": "You are not allowed to remove a team member"
              },
              "YOU_ARE_NOT_ALLOWED_TO_ACCESS_THIS_ORGANIZATION": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_ACCESS_THIS_ORGANIZATION",
                "message": "You are not allowed to access this organization as an owner"
              },
              "YOU_ARE_NOT_A_MEMBER_OF_THIS_ORGANIZATION": {
                "code": "YOU_ARE_NOT_A_MEMBER_OF_THIS_ORGANIZATION",
                "message": "You are not a member of this organization"
              },
              "MISSING_AC_INSTANCE": {
                "code": "MISSING_AC_INSTANCE",
                "message": "Dynamic Access Control requires a pre-defined ac instance on the server auth plugin. Read server logs for more information"
              },
              "YOU_MUST_BE_IN_AN_ORGANIZATION_TO_CREATE_A_ROLE": {
                "code": "YOU_MUST_BE_IN_AN_ORGANIZATION_TO_CREATE_A_ROLE",
                "message": "You must be in an organization to create a role"
              },
              "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_ROLE": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_CREATE_A_ROLE",
                "message": "You are not allowed to create a role"
              },
              "YOU_ARE_NOT_ALLOWED_TO_UPDATE_A_ROLE": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_UPDATE_A_ROLE",
                "message": "You are not allowed to update a role"
              },
              "YOU_ARE_NOT_ALLOWED_TO_DELETE_A_ROLE": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_DELETE_A_ROLE",
                "message": "You are not allowed to delete a role"
              },
              "YOU_ARE_NOT_ALLOWED_TO_READ_A_ROLE": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_READ_A_ROLE",
                "message": "You are not allowed to read a role"
              },
              "YOU_ARE_NOT_ALLOWED_TO_LIST_A_ROLE": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_LIST_A_ROLE",
                "message": "You are not allowed to list a role"
              },
              "YOU_ARE_NOT_ALLOWED_TO_GET_A_ROLE": {
                "code": "YOU_ARE_NOT_ALLOWED_TO_GET_A_ROLE",
                "message": "You are not allowed to get a role"
              },
              "TOO_MANY_ROLES": {
                "code": "TOO_MANY_ROLES",
                "message": "This organization has too many roles"
              },
              "INVALID_RESOURCE": {
                "code": "INVALID_RESOURCE",
                "message": "The provided permission includes an invalid resource"
              },
              "ROLE_NAME_IS_ALREADY_TAKEN": {
                "code": "ROLE_NAME_IS_ALREADY_TAKEN",
                "message": "That role name is already taken"
              },
              "CANNOT_DELETE_A_PRE_DEFINED_ROLE": {
                "code": "CANNOT_DELETE_A_PRE_DEFINED_ROLE",
                "message": "Cannot delete a pre-defined role"
              },
              "ROLE_IS_ASSIGNED_TO_MEMBERS": {
                "code": "ROLE_IS_ASSIGNED_TO_MEMBERS",
                "message": "Cannot delete a role that is assigned to members. Please reassign the members to a different role first"
              }
            },
            "options": {
              "allowUserToCreateOrganization": true,
              "organizationLimit": 50,
              "disableOrganizationDeletion": true,
              "creatorRole": "owner",
              "teams": {
                "enabled": true,
                "defaultTeam": {
                  "enabled": false
                },
                "allowRemovingAllTeams": false
              }
            }
          }
        },
        {
          "name": "api-key",
          "config": {
            "id": "api-key",
            "version": "1.6.2",
            "$ERROR_CODES": {
              "INVALID_METADATA_TYPE": {
                "code": "INVALID_METADATA_TYPE",
                "message": "metadata must be an object or undefined"
              },
              "REFILL_AMOUNT_AND_INTERVAL_REQUIRED": {
                "code": "REFILL_AMOUNT_AND_INTERVAL_REQUIRED",
                "message": "refillAmount is required when refillInterval is provided"
              },
              "REFILL_INTERVAL_AND_AMOUNT_REQUIRED": {
                "code": "REFILL_INTERVAL_AND_AMOUNT_REQUIRED",
                "message": "refillInterval is required when refillAmount is provided"
              },
              "USER_BANNED": {
                "code": "USER_BANNED",
                "message": "User is banned"
              },
              "UNAUTHORIZED_SESSION": {
                "code": "UNAUTHORIZED_SESSION",
                "message": "Unauthorized or invalid session"
              },
              "KEY_NOT_FOUND": {
                "code": "KEY_NOT_FOUND",
                "message": "API Key not found"
              },
              "KEY_DISABLED": {
                "code": "KEY_DISABLED",
                "message": "API Key is disabled"
              },
              "KEY_EXPIRED": {
                "code": "KEY_EXPIRED",
                "message": "API Key has expired"
              },
              "USAGE_EXCEEDED": {
                "code": "USAGE_EXCEEDED",
                "message": "API Key has reached its usage limit"
              },
              "KEY_NOT_RECOVERABLE": {
                "code": "KEY_NOT_RECOVERABLE",
                "message": "API Key is not recoverable"
              },
              "EXPIRES_IN_IS_TOO_SMALL": {
                "code": "EXPIRES_IN_IS_TOO_SMALL",
                "message": "The expiresIn is smaller than the predefined minimum value."
              },
              "EXPIRES_IN_IS_TOO_LARGE": {
                "code": "EXPIRES_IN_IS_TOO_LARGE",
                "message": "The expiresIn is larger than the predefined maximum value."
              },
              "INVALID_REMAINING": {
                "code": "INVALID_REMAINING",
                "message": "The remaining count is either too large or too small."
              },
              "INVALID_PREFIX_LENGTH": {
                "code": "INVALID_PREFIX_LENGTH",
                "message": "The prefix length is either too large or too small."
              },
              "INVALID_NAME_LENGTH": {
                "code": "INVALID_NAME_LENGTH",
                "message": "The name length is either too large or too small."
              },
              "METADATA_DISABLED": {
                "code": "METADATA_DISABLED",
                "message": "Metadata is disabled."
              },
              "RATE_LIMIT_EXCEEDED": {
                "code": "RATE_LIMIT_EXCEEDED",
                "message": "Rate limit exceeded."
              },
              "NO_VALUES_TO_UPDATE": {
                "code": "NO_VALUES_TO_UPDATE",
                "message": "No values to update."
              },
              "KEY_DISABLED_EXPIRATION": {
                "code": "KEY_DISABLED_EXPIRATION",
                "message": "Custom key expiration values are disabled."
              },
              "INVALID_API_KEY": {
                "code": "INVALID_API_KEY",
                "message": "Invalid API key."
              },
              "INVALID_USER_ID_FROM_API_KEY": {
                "code": "INVALID_USER_ID_FROM_API_KEY",
                "message": "The user id from the API key is invalid."
              },
              "INVALID_REFERENCE_ID_FROM_API_KEY": {
                "code": "INVALID_REFERENCE_ID_FROM_API_KEY",
                "message": "The reference id from the API key is invalid."
              },
              "INVALID_API_KEY_GETTER_RETURN_TYPE": {
                "code": "INVALID_API_KEY_GETTER_RETURN_TYPE",
                "message": "API Key getter returned an invalid key type. Expected string."
              },
              "SERVER_ONLY_PROPERTY": {
                "code": "SERVER_ONLY_PROPERTY",
                "message": "The property you're trying to set can only be set from the server auth instance only."
              },
              "FAILED_TO_UPDATE_API_KEY": {
                "code": "FAILED_TO_UPDATE_API_KEY",
                "message": "Failed to update API key"
              },
              "NAME_REQUIRED": {
                "code": "NAME_REQUIRED",
                "message": "API Key name is required."
              },
              "ORGANIZATION_ID_REQUIRED": {
                "code": "ORGANIZATION_ID_REQUIRED",
                "message": "Organization ID is required for organization-owned API keys."
              },
              "USER_NOT_MEMBER_OF_ORGANIZATION": {
                "code": "USER_NOT_MEMBER_OF_ORGANIZATION",
                "message": "You are not a member of the organization that owns this API key."
              },
              "INSUFFICIENT_API_KEY_PERMISSIONS": {
                "code": "INSUFFICIENT_API_KEY_PERMISSIONS",
                "message": "You do not have permission to perform this action on organization API keys."
              },
              "NO_DEFAULT_API_KEY_CONFIGURATION_FOUND": {
                "code": "NO_DEFAULT_API_KEY_CONFIGURATION_FOUND",
                "message": "No default api-key configuration found."
              },
              "ORGANIZATION_PLUGIN_REQUIRED": {
                "code": "ORGANIZATION_PLUGIN_REQUIRED",
                "message": "Organization plugin is required for organization-owned API keys. Please install and configure the organization plugin."
              }
            },
            "hooks": {
              "before": [
                {}
              ]
            },
            "endpoints": {},
            "schema": {
              "apikey": {
                "fields": {
                  "configId": {
                    "type": "string",
                    "required": true,
                    "defaultValue": "default",
                    "input": false,
                    "index": true
                  },
                  "name": {
                    "type": "string",
                    "required": false,
                    "input": false
                  },
                  "start": {
                    "type": "string",
                    "required": false,
                    "input": false
                  },
                  "referenceId": {
                    "type": "string",
                    "required": true,
                    "input": false,
                    "index": true
                  },
                  "prefix": {
                    "type": "string",
                    "required": false,
                    "input": false
                  },
                  "key": {
                    "type": "string",
                    "required": true,
                    "input": false,
                    "index": true
                  },
                  "refillInterval": {
                    "type": "number",
                    "required": false,
                    "input": false
                  },
                  "refillAmount": {
                    "type": "number",
                    "required": false,
                    "input": false
                  },
                  "lastRefillAt": {
                    "type": "date",
                    "required": false,
                    "input": false
                  },
                  "enabled": {
                    "type": "boolean",
                    "required": false,
                    "input": false,
                    "defaultValue": true
                  },
                  "rateLimitEnabled": {
                    "type": "boolean",
                    "required": false,
                    "input": false,
                    "defaultValue": true
                  },
                  "rateLimitTimeWindow": {
                    "type": "number",
                    "required": false,
                    "input": false,
                    "defaultValue": 86400000
                  },
                  "rateLimitMax": {
                    "type": "number",
                    "required": false,
                    "input": false,
                    "defaultValue": 10
                  },
                  "requestCount": {
                    "type": "number",
                    "required": false,
                    "input": false,
                    "defaultValue": 0
                  },
                  "remaining": {
                    "type": "number",
                    "required": false,
                    "input": false
                  },
                  "lastRequest": {
                    "type": "date",
                    "required": false,
                    "input": false
                  },
                  "expiresAt": {
                    "type": "date",
                    "required": false,
                    "input": false
                  },
                  "createdAt": {
                    "type": "date",
                    "required": true,
                    "input": false
                  },
                  "updatedAt": {
                    "type": "date",
                    "required": true,
                    "input": false
                  },
                  "permissions": {
                    "type": "string",
                    "required": false,
                    "input": false
                  },
                  "metadata": {
                    "type": "string",
                    "required": false,
                    "input": true,
                    "transform": {}
                  }
                }
              }
            }
          }
        },
        {
          "name": "sso",
          "config": {
            "id": "sso",
            "version": "1.6.2",
            "endpoints": {},
            "hooks": {
              "before": [
                {}
              ],
              "after": [
                {}
              ]
            },
            "schema": {
              "ssoProvider": {
                "modelName": "ssoProvider",
                "fields": {
                  "issuer": {
                    "type": "string",
                    "required": true,
                    "fieldName": "issuer"
                  },
                  "oidcConfig": {
                    "type": "string",
                    "required": false,
                    "fieldName": "oidcConfig"
                  },
                  "samlConfig": {
                    "type": "string",
                    "required": false,
                    "fieldName": "samlConfig"
                  },
                  "userId": {
                    "type": "string",
                    "references": {
                      "model": "user",
                      "field": "id"
                    },
                    "fieldName": "userId"
                  },
                  "providerId": {
                    "type": "string",
                    "required": true,
                    "unique": true,
                    "fieldName": "providerId"
                  },
                  "organizationId": {
                    "type": "string",
                    "required": false,
                    "fieldName": "organizationId"
                  },
                  "domain": {
                    "type": "string",
                    "required": true,
                    "fieldName": "domain"
                  },
                  "domainVerified": {
                    "type": "boolean",
                    "required": false
                  }
                }
              }
            },
            "options": {
              "domainVerification": {
                "enabled": true,
                "tokenPrefix": "ops-dv"
              },
              "organizationProvisioning": {
                "disabled": true,
                "defaultRole": "member"
              }
            }
          }
        },
        {
          "name": "last-login-method",
          "config": {
            "id": "last-login-method",
            "version": "1.6.2",
            "hooks": {
              "after": [
                {}
              ]
            },
            "options": {
              "cookieName": "ops.last_login"
            }
          }
        },
        {
          "name": "open-api",
          "config": {
            "id": "open-api",
            "version": "1.6.2",
            "endpoints": {},
            "options": {}
          }
        }
      ],
      "logger": {
        "disabled": false,
        "disableColors": false,
        "level": "debug"
      }
    }
  }
}
```

### Which area(s) are affected? (Select all that apply)

Backend, Client

### Auth config (if applicable)

```typescript
import { betterAuth } from "better-auth"
export const auth = betterAuth({
  emailAndPassword: {  
    enabled: true
  },
  plugins: [
    organization()
  ]
});
```

### Additional context

Also Something around this but not related to this bug, the `listUserTeams` returns all teams from all organizations. Is it possible to just just teams of user for active orgniazation? Is this intended or a bug as well? or I missed something

### PR DESCRIPTION
**Pull request by GautamBytes**
## Summary

- scope `organization.setActiveTeam` lookups to the current active organization
- return `NO_ACTIVE_ORGANIZATION` when there is no active org for team validation
- document the active-organization requirement and add regression coverage

## Root cause

`setActiveTeam` previously resolved teams by raw `teamId` and only checked team membership. That allowed a session to keep `activeOrganizationId` from one organization while setting `activeTeamId` from a different organization.

---
## Summary by cubic
Scopes `organization.setActiveTeam` to the session’s active organization and requires an active org before setting or refreshing a team. Updates docs/OpenAPI and adds org‑scoping tests; ships a patch to `better-auth`.

- **Bug Fixes**
  - Validate teams by `teamId` within the active `organizationId`; reject out‑of‑scope teams and keep the current active team.
  - Return `NO_ACTIVE_ORGANIZATION` when no active org is set; on org drift, reject refresh calls with `TEAM_NOT_FOUND`.
  - Clarify docs/OpenAPI (team must be in the active org; `teamId: string | null`).

- **Migration**
  - Set an active organization before calling `/organization/set-active-team`, or the API returns `NO_ACTIVE_ORGANIZATION`.

<sup>Written for commit 24a70cfc5e4dde8ddfc7df1be3f6285172cb3641. Summary will update on new commits.</sup>

### CODE REVIEW

**Review by ping-maxwell — COMMENTED** [review]
Consider this sequence:
1. User is in Org A with active team T1 (from Org A)
2. Some external process sets activeOrganizationId to Org B on the session
3. User calls `setActiveTeam({})` (no `teamId`, the intent is to "refresh" the current active team)

Before this PR, it would succeed and the user keeps T1, however after this PR, it will return null -> throws `TEAM_NOT_FOUND`. This is arguably the right behaviour (the team genuinely doesn't belong to the active org). but it's a subtle change in the no-teamId path that isn't covered by the new tests. A test for this edge case would prevent future regressions 🤔

File: packages/better-auth/src/plugins/organization/team.test.ts [hunk 1]
```diff
@@ -500,6 +501,199 @@ describe("team", async () => {
 	});
 });
 
+describe("setActiveTeam org scoping", async () => {
```
Comment (ping-maxwell) [inline packages/better-auth/src/plugins/organization/team.test.ts:508]: please add 
```
/**
 * @see https://github.com/better-auth/better-auth/issues/9237
 */
```

**Review by ping-maxwell — CHANGES_REQUESTED** [review]

**Review by ping-maxwell — CHANGES_REQUESTED** [review]
fix CI and this should be good!

**Review by ping-maxwell — APPROVED** [review]
